pkg/apis/core/v1alpha1/storagecluster.go

```diff
@@ -0,0 +1,265 @@
+	// SystemMdDevice device that will be used to store system metadata
+	SystemMdDevice *string `json:"systemMetadataDevice"`
+	// DataStorageType backing store type for managing drives and pools
+	DataStorageType *string `json:"dataStorageType"`
+	// RaidLevel raid level for the storage pool
+	RaidLevel *string `json:"raidLevel"`
+}
+
+// CloudStorageSpec details of storage in cloud environment
+type CloudStorageSpec struct {
+	// DeviceSpecs list of storage device specs. A cloud storage device will
+	// be created for every spec in the DeviceSpecs list. Currently,
+	// CloudStorageSpec is only at the cluster level, so the below specs
+	// be applied to all storage nodes in the cluster.
+	DeviceSpecs *[]string `json:"deviceSpecs"`
+	// JournalDeviceSpec spec for the journal device
+	JournalDeviceSpec *string `json:"journalDeviceSpec"`
+	// SystemMdDeviceSpec spec for the metadata device
+	SystemMdDeviceSpec *string `json:"systemMetadataDeviceSpec"`
+	// MaxStorageNodes maximum nodes that will have storage in the cluster
+	MaxStorageNodes uint32 `json:"maxStorageNodes"`
+	// MaxStorageNodesPerZone maximum nodes in every zone that will have
+	// storage in the cluster
+	MaxStorageNodesPerZone uint32 `json:"maxStorageNodesPerZone"`
+}
+
+// Geography is topology information for a node
+type Geography struct {
+	// Region region in which the node is placed
+	Region string `json:"region"`
+	// Zone zone in which the node is placed
+	Zone string `json:"zone"`
+	// Rack rack on which the node is placed
+	Rack string `json:"rack"`
+}
+
+// StorageClusterStatus is the status of a storage cluster
+type StorageClusterStatus struct {
+	// ClusterName name of the storage cluster
+	ClusterName string `json:"clusterName"`
+	// ClusterUUID uuid for the storage cluster
+	ClusterUUID string `json:"clusterUuid"`
+	// CreatedAt timestamp at which the storage cluster was created
+	CreatedAt *meta.Time `json:"createdAt"`
+	// Status status of the storage cluster
+	Status ClusterStatus `json:"status"`
+	// Reason is human readable message indicating the status of the cluster
+	Reason string `json:"reason,omitempty"`
+	// NodeStatuses list of statuses for all the nodes in the storage cluster
+	NodeStatuses []NodeStatus `json:nodes`
+}
+
+// ClusterStatus is the enum type for cluster statuses
+type ClusterStatus string
+
+// These are valid cluster statuses.
+const (
+	// ClusterOK means the cluster is up and healthy
+	ClusterOk ClusterStatus = "Ok"
+	// ClusterOffline means the cluster is offline
+	ClusterOffline ClusterStatus = "Offline"
+	// ClusterNotInQuorum means the cluster is out of quorum
+	ClusterNotInQuorum ClusterStatus = "NotInQuorum"
+	// ClusterUnknown means the cluser status is not known
+	ClusterUnknown ClusterStatus = "Unknown"
+)
+
+// NodeStatus status of the storage cluster node
+type NodeStatus struct {
+	// NodeName name of the node
+	NodeName string `json:"nodeName"`
+	// NodeUUID uuid of the node
+	NodeUUID string `json:"nodeUuid"`
```

Comment:
Does that need to be `nodeUUID`, which is the convention for other fields like `dataIP`?